autopwn.py

```diff
@@ -45,47 +35,76 @@ def StartScanning(
         Targets = [targetarg]
 
     ScanPorts, ScanVulns, DownloadExploits = UserConfirmation()
-    ScanWeb = WebScan()
+    PortArray = []
+
+    all_vulnerabilities = []
 
     for host in Targets:
         if ScanPorts:
             PortScanResults = PortScan(
-                host, log, args.speed, args.host_timeout, scanmode, args.nmap_flags
+                host, log, args.speed, args.host_timeout, scanmode, args.nmap_flags, shodan_api_key
             )
-            PortArray = AnalyseScanResults(PortScanResults, log, console, host)
-            if ScanVulns and len(PortArray) > 0:
-                VulnsArray = SearchSploits(PortArray, log, console, console2, apiKey)
-                if DownloadExploits and len(VulnsArray) > 0:
-                    GetExploitsFromArray(VulnsArray, log, console, console2, host)
-
-        if ScanWeb:
-            webvuln(host, log, console)
+            PortArray = AnalyseScanResults(PortScanResults, log, console, host, shodan_results=None)
+        if ScanVulns and PortArray and len(PortArray) > 0:
+            VulnsArray = []
+            keywords = generate_keywords(PortArray)
+            sploits = SearchSploits(keywords, log, console, args, apiKey)
+            for sploit in sploits:
+                all_vulnerabilities.append(sploit)
+            if shodan_api_key:
+                ShodanVulns, ShodanPorts = GetShodanVulns(host, shodan_api_key, log)
+                for port in ShodanPorts:
+                    PortArray.append((host, port, "tcp", "shodan", ""))
+                for vuln in ShodanVulns:
+                    log.logger("info", f"Shodan Vuln: {vuln['title']} - CVEs: {', '.join(vuln['CVEs'])}")
+                    vuln_obj = VulnerableSoftware(
+                        title=vuln['title'],
+                        CVEs=vuln['CVEs'],
+                        severity_score=vuln['severity_score'],
+                        exploitability=vuln['exploitability']
+                    )
+                    all_vulnerabilities.append(vuln_obj)
+            if zoomeye_api_key:
+                ZoomEyeVulns = GetZoomEyeVulns(host, zoomeye_api_key, log)
+                for vuln in ZoomEyeVulns:
+                    vuln_obj = VulnerableSoftware(
+                        title=vuln['title'],
+                        CVEs=vuln['CVEs'],
+                        severity_score=vuln['severity_score'],
+                        exploitability=vuln['exploitability']
+                    )
+                    all_vulnerabilities.append(vuln_obj)
+                    
+            all_vulnerabilities = remove_duplicate_vulnerabilities(all_vulnerabilities)
+            if all_vulnerabilities:
+                print("All vulnerabilities: ", all_vulnerabilities)
```

Comment:
This way of printing doesn't fit programs way of printing vulnerabilities and should be changed to print like SearchSploits function in modules.searchvuln